deploy/awstools/awstools.py

```diff
@@ -191,6 +191,9 @@ def launch_instances(instancetype, count, instancemarket, spotinterruptionbehavi
         When true, create `count` instances, regardless of whether any already exist. When False, only
         create instances until there are `count` total instances that match `tags` and `instancetype`
         If `tags` are not passed, `always_expand` must be `True` or `ValueError` is thrown.
+    ami_id : str, default=None
+        Override AMI ID to use for launching instances. `None` results in the default AMI ID specified by
+        `awstools.get_f1_ami_id()`.
```

Comment:
I don't know why I didn't see this and comment the first time around.  Feel feel free to ignore these suggestions but it might be more straightforward to just do:
```suggestion
    ami_id : str, default=awstools.get_f1_ami_id()
       AMI ID to use for launching instances.
```
**And you'd also need to change the signature and remove line 230.**

If you don't like this suggestion, then you probably need to change the type in the docstring to `Optional[str]` or we can just do that when we switch this over to using actual type hints that are checked by mypy.

Reply:
Doing something like this results in the function being called whenever the `awstools` module is imported (and erroring if the AWS env. variables aren't set). For now, I'll just keep this an option and we can clean this up more later.